azurerm/resource_arm_batch_account.go

```diff
@@ -0,0 +1,224 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/services/batch/mgmt/2017-09-01/batch"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmBatchAccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmBatchAccountCreate,
+		Read:   resourceArmBatchAccountRead,
+		Update: resourceArmBatchAccountUpdate,
+		Delete: resourceArmBatchAccountDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMBatchAccountName,
+			},
+			"resource_group_name": resourceGroupNameDiffSuppressSchema(),
```

Comment:
(as above) we should use the case sensitive name here
```suggestion
			"resource_group_name": resourceGroupNameSchema(),
```